code/modules/projectiles/projectile/magic.dm

```diff
@@ -158,7 +158,8 @@
 						if("drone")
 							new_mob = new /mob/living/simple_animal/drone(M.loc)
 							var/mob/living/simple_animal/drone/D = new_mob
-							D.update_drone_hack()
+							spawn(50) //this should give us time for the rest of the proc to finish so a mob will be in the drone
```

Comment:
Yeah, this proc is awful and full of spaghetti and everything sucks, but oranges' solution sucks slightly less than yours.